openfisca_uk/variables/finance/benefit/family/LHA.py

```diff
@@ -76,12 +92,42 @@ def formula(benunit, period, parameters):
                 num_rooms > 3,
             ],
             [
-                rates.shared,
-                rates.beds_1,
-                rates.beds_2,
-                rates.beds_3,
-                rates.beds_4,
+                LHACategory.A,
+                LHACategory.B,
+                LHACategory.C,
+                LHACategory.D,
+                LHACategory.E,
             ],
         )
-        amount = min_(rent, rate_cap)
-        return amount
+        return category
+
+
+class BRMA_LHA_rate(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"LHA Rate"
+    definition_period = WEEK
+
+    def formula(benunit, period, parameters):
+        personal_BRMA = benunit.members.household(
+            "BRMA", period.this_year
+        ).decode_to_str()
+        BRMA = (
```

Comment:
Am I interpreting this right as selecting the BRMA of the first member of the benefit unit? Could you add a comment on this?

Reply:
Yeah, this was a bit complicated. Essentially, we need to transpose a household's BRMA onto each benefit unit because housing benefit is at the family level, rather than household level. Usually we can just project to person-level (person_BRMA = `benunit.members.household("BRMA")`) and then aggregate to benefit unit level (`family_BRMA = benunit.value_from_first_person(person_BRMA)`). But this doesn't work for the Enum type, which gets stuck as the number, and the decode function is lost somehow.

Reply:
Gotcha, could you add a comment to the code? Should there be an issue in openfisca-uk or openfisca-core for the projecting and aggregating enums?

Reply:
Gotcha, could you add a comment to the code? Should there be an issue in openfisca-uk or openfisca-core for the projecting and aggregating enums?